core/rawdb/schema.go

```diff
@@ -188,7 +188,11 @@ func accountSnapshotKey(hash common.Hash) []byte {
 
 // storageSnapshotKey = SnapshotStoragePrefix + account hash + storage hash
 func storageSnapshotKey(accountHash, storageHash common.Hash) []byte {
-	return append(append(SnapshotStoragePrefix, accountHash.Bytes()...), storageHash.Bytes()...)
+	buf := make([]byte, len(SnapshotStoragePrefix)+common.HashLength+common.HashLength)
+	n := copy(buf, SnapshotStoragePrefix)
+	n += copy(buf[n:], accountHash.Bytes())
+	copy(buf[n:], storageHash.Bytes())
+	return buf
```

Comment:
I _suspect_ that a more minimal change would have the same effect

```suggestion
	buf := make([]byte, len(SnapshotStoragePrefix)+common.HashLength+common.HashLength)
	return append(append(append(buf, SnapshotStoragePrefix...), accountHash.Bytes()...), storageHash.Bytes()...)
```
It is slightly less complex since it relies on the compiler instead of fiddling with `n`.